.github/workflows/test.yml

```diff
@@ -186,6 +189,47 @@ jobs:
       - name: Publish code coverage
         uses: codecov/codecov-action@v4
 
+  detect:
+    needs: test-nosoftdeps
+    name: detect changes
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      datasets: ${{ steps.filter.outputs.datasets }}
+      pyproject: ${{ steps.filter.outputs.pyproject }}
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            datasets:
+              - sktime/base/**
+              - sktime/datasets/**
+            pyproject:
+              - pyproject.toml
```

Comment:
I needed it for testing extras can be installed or not, but you removed it. This can possibly be removed.

Reply:
hm, can you explain? I don't quite get your statement here.

Reply:
I don't think you are using "detection of change in pyproject.toml" anywhere after your modifications. So my guess is removing these two lines, plus outset creation to store result of that detection check is not necessary anymore.

Reply:
Ah, I see. You are right.

May I suggest we leave it in for now, as this is not the final step of rework?
E.g., one may like to move some parts of the "detects"-s further down into that one step. Or, trigger extensive tests only if `pyproject` changes, etc.

Pending of course discussion of how we reduce the footprint of the "old tests".